assignments/callbacks.js

```diff
@@ -2,23 +2,52 @@ const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum'];
 
 function firstItem(arr, cb) {
   // firstItem passes the first item of the given array to the callback function.
+  return cb(arr);
 }
+function first(items) {
+  return console.log(items[0]);
+}
+firstItem(items, first)
+
```

Comment:
So the instructions ask that `firstItem` pass the first item of its array into the callback function given to it. In practice this would look more like this:

 ```
function firstItem(arr, cb) {
  // firstItem passes the first item of the given array to the callback function.
  return cb(arr[0]);
}
```
For this exercise there isn't a specified callback function, but a similar correction needs to happen on all the future exercises as well. The way you have it structured now, the callback executing functions make no change and serve no purpose. Does that make sense? You want the logic to be inside of firstItem, not in a separate function. So, for example, you could have a simple print function. Something like: 
```
function print(output){
   console.log(output);
}
```
That you pass in as you "cb" for all the exercises. If you did this for first item, then your result would be to print the first item in the array you pass in. You'd invoke it like this:
```
firstItem(items, print);
```
And your output would be: `Pencil`

Try to go back through and restructure your callbacks. If you are still confused, let me know!

Reply:
If my brain hasn't melted too far out of my head at the end of today, I'll definitely take a look over my callbacks. Sadly, this is often the case and I find the weekend a better time to review my previous work. Here's to hoping today goes much better!